orttraining/orttraining/test/graph/gradient_graph_builder_test.cc

```diff
@@ -1115,7 +1115,113 @@ void RetrieveSendRecvOperators(
   }
 }
 
-TEST(GradientGraphBuilderTest, PipelineOnlinePartition) {
+TEST(GradientGraphBuilderTest, PipelineOnlinePartition_bert_tiny) {
+  const auto model_path = ORT_TSTR("testdata/bert-tiny.onnx");
+
+  const size_t total_partition_count = 3;
+  TrainingSession::TrainingConfiguration::PipelineConfiguration pipe{};
+  pipe.do_partition = true;
+
+  // evenly model in 3 partitions
+  TrainingSession::TrainingConfiguration::CutInfo cut0 = {
+      onnxruntime::training::TrainingSession::TrainingConfiguration::CutEdge("186"),
+      onnxruntime::training::TrainingSession::TrainingConfiguration::CutEdge("71", {"273"})};
+
+  TrainingSession::TrainingConfiguration::CutInfo cut1 = {
+      onnxruntime::training::TrainingSession::TrainingConfiguration::CutEdge("308"),
+      onnxruntime::training::TrainingSession::TrainingConfiguration::CutEdge("71", {"395"})};
+
+  pipe.cut_list.emplace_back(cut0);
+  pipe.cut_list.emplace_back(cut1);
+
+  TrainingSession::TrainingConfiguration::MixedPrecisionConfiguration mixed_precision_config{};
+  mixed_precision_config.use_fp16_initializers = true;
+
+  // 2 test variations - full precision and mixed precision
+  const std::vector<bool> test_with_fp32{true, false};
+  for (auto is_fp32 : test_with_fp32) {
+    // graph is partitioned into 4 parts.
```

Comment:
3 parts?

Reply:
Right, corrected. (was trying 4 parts for a different model : ) )

---
In reply to: [449310982](https://github.com/microsoft/onnxruntime/pull/4321#discussion_r449310982) [](ancestors = 449310982)